base/ryu/Ryu.jl

```diff
@@ -17,6 +17,15 @@ neededdigits(::Type{Float64}) = 309 + 17
 neededdigits(::Type{Float32}) = 39 + 9 + 2
 neededdigits(::Type{Float16}) = 9 + 5 + 9
 
+"""
+    Ryu.neededfloatdigits(T)
+
+Number of digits necessary to represent type `T` in shortest precision.
+"""
+neededfloatdigits(::Type{Float64}) = 23
+neededfloatdigits(::Type{Float32}) = 15
```

Comment:
9 + decimal point + sign bit + "f" + second sign bit + max exponent (2) = 15

Reply:
```
julia> extrema(i->length(string(reinterpret(Float32, i))), typemin(UInt32):typemax(UInt32))
(3, 15)
```

Reply:
```julia-repl
julia> length(sprint(show, -0.000110974266f0))
17
```

Reply:
```suggestion
neededfloatdigits(::Type{Float32}) = 17
```
Add 2 for `0f` (see https://github.com/JuliaLang/julia/pull/57977#pullrequestreview-3059472408)